cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/RunnableUtils.java

```diff
@@ -203,6 +209,28 @@ public static void sanityCheckBrokersHavingOfflineReplicasOnBadDisks(List<String
     }
   }
 
+  /**
+   * Gracefully stop the ongoing execution (if any) and wait until the execution stops.
+   *
+   * @param kafkaCruiseControl The Kafka Cruise Control instance.
+   */
+  public static void maybeStopOngoingExecutionToModifyAndWait(KafkaCruiseControl kafkaCruiseControl) {
+    if (kafkaCruiseControl.hasOngoingExecution() && kafkaCruiseControl.modifyOngoingExecution(true)) {
+      LOG.info("Gracefully stopping the ongoing execution... Use {} endpoint with {}=true to force-stop it now.",
+               STOP_PROPOSAL_EXECUTION, FORCE_STOP_PARAM);
+      while (kafkaCruiseControl.executionState() != NO_TASK_IN_PROGRESS) {
+        // Stop ongoing execution, and wait for the execution to stop.
+        try {
```

Comment:
The wait here could be a very long time(or even forever since we do not force stop here), maybe we should assign a [`OperationStep`](https://github.com/linkedin/cruise-control/blob/472173237d96dbe7e47bbd16672ee45472706921/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/progress/OperationStep.java) to it, so user know that Cruise Control is current wait for ongoing task to finish.